proposals/eligibility_validator.go

```diff
@@ -145,8 +146,8 @@ func (v *Validator) CheckEligibility(ctx context.Context, ballot *types.Ballot)
 	if err != nil {
 		return false, err
 	}
-	if ballot.EligibilityCount != numEligibleSlots {
-		return false, fmt.Errorf("%w: expected %v, got: %v", errIncorrectEligCount, numEligibleSlots, ballot.EligibilityCount)
+	if advertisedCount != numEligibleSlots {
+		return false, fmt.Errorf("%w: expected %v, got: %v", errIncorrectEligCount, numEligibleSlots, advertisedCount)
```

Comment:
in that case it seems we just need to verify it once with ref ballot and not the following ballots.